en/docs/deploy/deploy-your-component.md

```diff
@@ -4,23 +4,97 @@ Deploying your component on Choreo makes it invocable. Once you have designed, t
+
+- Be sure that the implementation in the `.bal` file matches the component type. For example, if the component is a REST API, the `*.bal` file should contain an implementation for a REST API.
+- Include other dependencies for the Ballerina code. For more information, see [Ballerina Documentation- Organiza Ballerina Code](https://ballerina.io/learn/organize-ballerina-code/)
+
+## Configure variables
+
+A configurable variable is a parameter for which you specify the value when you deploy the component that includes it.
+
+A component implementation may include configurable variables in the following scenarios:
+
+- When a component is accessible by multiple people in an organization, you may want to avoid exposing some of the parameter values (that you consider sensitive information) in the component implementation.
+
+  For example, consider a REST API that returns a password to users whose requests meet specific conditions. In this scenario, you may not want to expose that password to everyone who can access the REST API implementation.
+
+- Each component user needs to provide a different value for a specific parameter.
+
+  For example, each component user may want to provide a unique email address to receive the response.
+
+When deploying a component with configurable variables, you need to click **Configure & Deploy** and then provide values for the configurable variables in the **Configure & Deploy** pane before deploying the component.
+
+If you want to edit the values you entered for the configurable variables after deploying the component, you can click **Configurables** on the **Development** card and make the required changes in the **Configure & Deploy** pane.
+
+![Edit configurables](../assets/img/deploy/edit-configurables.png){.cInlineImage-full}
+
+To learn how to define configurable variables, see [Ballerina Documentation -  Configurable Variables](https://ballerina.io/learn/by-example/configurable-variables/).
+
+## Deploy automatically
+
+Choreo enables automatic deployment for each component by default to automatically deploy it whenever a change is committed to the GitHub repository in which its implementation resides. You can disable it if required. However, Choreo requires you to perform the first deployment manually so that you can provide values for any [configurable variables](#configure-variables) that the component implementation may include.
+
+An automatic deployment can fail due to one or more of the following reasons:
+
+- If the code of the component implementation is invalid after the last change.
+- If you added new configurable variables to the component implementation.
+- If you deleted/changed one or more package dependencies (for more information, see [Ballerina Documentation- Organiza Ballerina Code](https://ballerina.io/learn/organize-ballerina-code/)).
+
+## Enter sandbox endpoint
+
+A sandbox environment is a test environment where you can test a component before pushing it to production.
```

Comment:
Since we currently only show 2 environments as Development and Production and sandbox is part of these, shall we avoid emphasizing on  word  _sandbox **environment**_